solutions/usaco-572.mdx

```diff
@@ -52,55 +58,79 @@ int main() {
 ### Implementation
 
 ```java
-import java.io.File;
-import java.io.PrintWriter;
-import java.util.Scanner;
-
-public class bcount {
-	public static void main(String[] args) throws Exception {
-		Scanner sc = new Scanner(new File("bcount.in"));
-		PrintWriter out = new PrintWriter("bcount.out");
-
-		int N = sc.nextInt();
-		int Q = sc.nextInt();
-		int[] Holsteins = new int[N + 1];
-		int[] Guernseys = new int[N + 1];
-		int[] Jerseys = new int[N + 1];
-
-		for (int i = 1; i < N + 1; i++) {
-			Holsteins[i] += Holsteins[i - 1];
-			Guernseys[i] += Guernseys[i - 1];
-			Jerseys[i] += Jerseys[i - 1];
-
-			int breed = sc.nextInt();
-
-			if (breed == 1) {
-				Holsteins[i] ++;
-			} else if (breed == 2) {
-				Guernseys[i] ++;
-			} else if (breed == 3) {
-				Jerseys[i] ++;
-			} else {
-				System.out.println("What the...");
-			}
+import java.io.*;
+import java.util.*;
+
+public class BCount {
+	public static void main(String[] args) throws IOException {
+		BufferedReader read = new BufferedReader(new FileReader("bcount.in"));
+		PrintWriter out = new PrintWriter(new BufferedWriter(new FileWriter("bcount.out")));
+		StringTokenizer initial = new StringTokenizer(read.readLine());
+		int cowNum = Integer.parseInt(initial.nextToken());
+		int queryNum = Integer.parseInt(initial.nextToken());
+		
+		int[] holsteins = new int[cowNum + 1];
+		int[] guernseys = new int[cowNum + 1];
+		int[] jerseys = new int[cowNum + 1];
+		for (int i = 1; i <= cowNum; i++) {
+			holsteins[i] += holsteins[i - 1];
+			guernseys[i] += guernseys[i - 1];
+			jerseys[i] += jerseys[i - 1];
+			
+			int breed = Integer.parseInt(read.readLine());
+			if (breed == 1)
+				holsteins[i]++;
+			if (breed == 2)
+				guernseys[i]++;
+			if (breed == 3)
```

Comment:
@andrewwangva why not else if?

Reply:
whoops idk i just felt it was more concise + there's no difference

Reply:
i think there is a difference?

Reply:
hmm how? i think its the same logically since breed can't be both 1 and 2?

Reply:
yeah, so isn't it *technically* more efficient to use else if? so breed == 2 and breed == 3 don't have to be checked if breed == 1

Reply:
true idk both are perfectly fine just kinda an impulsive change to make it look clean to me :)